### PR TITLE
setcoef now validates in gt4py

### PR DESCRIPTION
**Pull request by andrewpauling**
setcoef from the SW scheme is ported and validates against the serialized Fortran output in gt4py

### CODE REVIEW

File: radiation/python/radsw/setcoef_gt4py.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    plog: FIELD_FLT,
+    fp: FIELD_FLT,
+    fp1: FIELD_FLT,
+    ft: FIELD_FLT,
+    ft1: FIELD_FLT,
+    tem1: FIELD_FLT,
+    tem2: FIELD_FLT,
+    jp1: FIELD_INT,
+    preflog: Field[(DTYPE_FLT, (59,))],
+    tref: Field[(DTYPE_FLT, (59,))],
```
Comment (rheacangeo) [inline radiation/python/radsw/setcoef_gt4py.py:123]: what is '59', can we use a variable for this?

File: radiation/python/radsw/setcoef_gt4py.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        jt1 = max(1, min(4, 3.0 + tem2)) - 1
+        ft = tem1 - (jt - 2)
+        ft1 = tem2 - (jt1 - 2)
+
+        #  --- ...  we have now isolated the layer ln pressure and temperature,
+        #           between two reference pressures and two reference temperatures
+        #           (for each reference pressure).  we multiply the pressure
+        #           fraction fp with the appropriate temperature fractions to get
+        #           the factors that will be needed for the interpolation that yields
+        #           the optical depths (performed in routines taugbn for band n).
```
Comment (rheacangeo) [inline radiation/python/radsw/setcoef_gt4py.py:159]: great including the comments!

File: radiation/python/radsw/setcoef_gt4py.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        fac00 = fp1 * (1.0 - ft)
+        fac11 = fp * ft1
+        fac01 = fp * (1.0 - ft1)
+
+        #  --- ...  if the pressure is less than ~100mb, perform a different
+        #           set of species interpolations.
+
+        if plog > 4.56:
+
+            laytrop = True
```
Comment (rheacangeo) [inline radiation/python/radsw/setcoef_gt4py.py:172]: could provide a comment saying something about how this is for levels in the troposphere. it seems obvious to me, but could help someone unfamiliar

**Review by rheacangeo — APPROVED** [review]
Looks great! one minor optional comment!

File: radiation/python/radsw/setcoef_gt4py.py (hunk 1 above)
Comment (andrewpauling) [reply radiation/python/radsw/setcoef_gt4py.py:123]: Unfortunately not, they are hardcoded in the Fortran too

File: radiation/python/radsw/setcoef_gt4py.py (hunk 3 above)
Comment (andrewpauling) [reply radiation/python/radsw/setcoef_gt4py.py:172]: Sure will do

**Pull request merged**